### PR TITLE
Use custom review page for Simple Forms form upload tool

### PR DESCRIPTION
**Pull request by Thrillberg**
## Summary
This PR removes the defunct review page in favor of the custom one that is already in the repo. It also adds the Edit links and modifies the copy a bit.

~~If possible, I'd like attention paid to the Edit links since I think when they are clicked, `formData` is lost. Is there a way to retain `formData` across the link?~~

## Related issue(s)
https://app.zenhub.com/workspaces/vagov-product-team-forms-634853151f5c6000165942bc/issues/gh/department-of-veterans-affairs/va.gov-team-forms/1518

## Screenshots
Screenshot 2024-08-13 at 10 44 11 AM

### CODE REVIEW

File: src/applications/simple-forms/form-upload/containers/CustomReviewTopContent.jsx [hunk 1]
```diff
@@ -41,42 +35,46 @@ const CustomReviewTopContent = () => {
...
     <>
-      <div className="vads-u-margin-top--2">
-        <div className="vads-u-margin-y--1 vads-u-color--gray">Your file</div>
-        {uploadedFile && renderFileInfo(uploadedFile)}
-      </div>
-      <div className="vads-u-border-bottom--1px vads-u-margin-top--1 vads-u-margin-bottom--4">
-        <h3>Your personal information</h3>
+      <div className="vads-u-display--flex vads-l-row vads-u-justify-content--space-between vads-u-align-items--baseline vads-u-border-bottom--1px vads-u-margin-top--1 vads-u-margin-bottom--4">
+        <h4>Personal information</h4>
+        <VaLink
```
Comment (Thrillberg) [inline src/applications/simple-forms/form-upload/containers/CustomReviewTopContent.jsx:60]: Will clicking this link cause `formData` to be lost? That seems like a bug.

File: src/applications/simple-forms/form-upload/containers/EditLink.jsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  function onClick(e) {
+    e.preventDefault();
+    router.push(link);
+  }
+
+  return <VaLink href={href} onClick={onClick} text="Edit" />;
+};
+
+function mapStateToProps(state) {
```
Comment (rhasselle-oddball) [inline src/applications/simple-forms/form-upload/containers/EditLink.jsx:25]: can get rid of this block and the connect wrapper

File: src/applications/simple-forms/form-upload/containers/EditLink.jsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router';
+import { VaLink } from '@department-of-veterans-affairs/component-library/dist/react-bindings';
+
+const EditLink = ({ href, router }) => {
+  const [link, setLink] = useState(null);
+
+  useEffect(
```
Comment (rhasselle-oddball) [inline src/applications/simple-forms/form-upload/containers/EditLink.jsx:10]: can get rid of the useEffect and useState I think and just set the href directly.

File: src/applications/simple-forms/form-upload/containers/EditLink.jsx (hunk 2 above)
Comment (Thrillberg) [reply src/applications/simple-forms/form-upload/containers/EditLink.jsx]: It looks like we do need this for the props (`router`) to get through.

**Review by rhasselle-oddball — DISMISSED** [review]

**Review by rhasselle-oddball — APPROVED** [review]

**Pull request merged**